docs/migration/migrate_to_new_azure_service_broker.md

```diff
@@ -0,0 +1,106 @@
+
+* The Azure environment of meta-azure-service-broker is public Azure (as open-service-broker-azure only supports public Azure currently). 
+
+* The meta-azure-service-broker version in using is equal or greater than 1.4.0.
+
+* You have succeeded to `cf push` [open-service-broker-azure](https://github.com/Azure/open-service-broker-azure) and haven't used `cf create-service-broker` to register it as service broker.
+
+* There is no existing instance for the services listed:
+  * azure-mysqldb
+  * azure-postgresqldb
+  * azure-cosmosdb
+  * azure-documentdb
+
+* Have NodeJS 6 or greater installed. Have NPM 3.x installed.
+
+* Have Golang 1.8.3 installed.
+
+## Migration Steps
+
+### 1. If you have azure-sqldb instances which are not on an existing server, prepare a list of their instance GUIDs.
+
+With the format in a plain text file:
+
+```
+d008a572-1ded-45b0-a2fe-4bb0fdedb645
+325aedac-c2c3-47f4-a8fa-0b6dc343f94f
+6641fdfb-abef-4303-ad29-f055c1ca66cc
+```
+
+>>Tip: The CF CLI command `cf service <instance-name> --guid` can check the GUID of a service instance.
+
+For the azure-sqldb instances without the list, the migration script will treat them as existing server case.
+
+### 2. Purge services offering via CF CLI
+
+As above listed services are not able to migrate. We need to purge them via CF CLI. run:
+
+```
+$ cf purge-service-offering azure-mysqldb -f
+$ cf purge-service-offering azure-postgresqldb -f
+$ cf purge-service-offering azure-cosmosdb -f
+$ cf purge-service-offering azure-documentdb -f
+```
+
+>>**WARNING**: `cf purge-service-offering SERVICE` assumes that the service broker responsible for this service offering is no longer available, and all service instances have been deleted. The resources of these instances are still on Azure. You can manually migrate them.
+
+### 3. Migrate service instances records from meta-azure-service-broker database(the SQL one) to open-service-broker-azure database(the redis one).
+
+1. Download migration scripts:
+
+```
+curl -L -O https://raw.githubusercontent.com/Azure/meta-azure-service-broker/master/docs/migration/migrate_to_new_azure_service_broker.sh
+curl -L -O https://raw.githubusercontent.com/Azure/meta-azure-service-broker/master/docs/migration/migrate_to_new_azure_service_broker.js
+curl -L -O https://raw.githubusercontent.com/Azure/meta-azure-service-broker/master/docs/migration/migrate_to_new_azure_service_broker_helper_pc.go
+curl -L -O https://raw.githubusercontent.com/Azure/meta-azure-service-broker/master/docs/migration/migrate_to_new_azure_service_broker_helper_bc.go
+```
+
+2. Get the `manifest.yml`(filled for `cf push`) of both meta-azure-service-broker and open-service-broker-azure.
+
+3. Run:
+
+```
+$ chmod +x migrate_to_new_azure_service_broker.sh
+$ ./migrate_to_new_azure_service_broker.sh <path-to-masb-manifest> <path-to-asb-manifest> (<path-to-new-sql-server-list>)
+```
+
+The broker database migration succeeds if the script ends with no error message.
```

Comment:
How to deal with the instances which failed to migrate?

Reply:
Will add a link to troubleshooting below.